### PR TITLE
[add] Form UI for the login component

### PR DESCRIPTION
**Pull request by edbella**
This initiates the addition of the Form View to the login component

### CODE REVIEW

File: src/components/signin/actions.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
-import axios from 'axios';
 import * as actionTypes from './actionTypes';
 
+// eslint-disable-next-line import/order
+import axios from 'axios';
+
 const {
     LOGIN_REQUEST, LOGIN_FAILURE, LOGOUT_FAILURE, LOGOUT_REQUEST,
```
Comment (justiceotuya) [inline src/components/signin/actions.js]: add all to one line or they should be in seperate lines 
const {
LOGIN_REQUEST, 
LOGIN_FAILURE, 
LOGOUT_FAILURE, 
LOGOUT_REQUEST,
...
}

File: src/components/signin/components/AntdForm.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import { Form, Icon, Input } from 'antd';
+import PropTypes from 'prop-types';
+
+class SignInForm extends Component {
+  handleSubmit = e => {
+      e.preventDefault();
+      const { form } = this.props;
+      form.validateFields((err, values) => {
+          if (!err) {
+              console.log('Received values of form: ', values);
```
Comment (justiceotuya) [inline src/components/signin/components/AntdForm.jsx]: no console.log. delete

File: src/components/signin/actions.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
-import axios from 'axios';
 import * as actionTypes from './actionTypes';
 
+// eslint-disable-next-line import/order
+import axios from 'axios';
+
 const {
```
Comment (justiceotuya) [inline src/components/signin/actions.js]: add all to one line or they should be in separate lines 
```
const {
LOGIN_REQUEST, 
LOGIN_FAILURE, 
LOGOUT_FAILURE, 
LOGOUT_REQUEST,
...
}
```

File: src/components/signin/components/PageHelmet.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    CANON_LINK,
+} = HELMET_STRINGS;
+
+export default function PageHelmet() {
+    return (
+        <Helmet
+            titleTemplate={TITLE_TEMPLATE}
+            defaultTitle={TITLE_TEMPLATE_DEFAULT_TITLE}
+        >
+            <meta charSet={CHARSET} />
```
Comment (justiceotuya) [inline src/components/signin/components/PageHelmet.jsx]: remember dont repeat yourself, fix the meta reptitition

**Comment by edbella** [comment]
@justiceotuya @olajohn-ajiboye @enyeInc Please review

File: src/components/signin/actions.js [hunk 5]
```diff
@@ -1,9 +1,16 @@
-import axios from 'axios';
 import * as actionTypes from './actionTypes';
 
+// eslint-disable-next-line import/order
```
Comment (enyeInc) [inline src/components/signin/actions.js]: why are you disabling this? Just import in the correct order.

File: src/components/signin/components/FormView.jsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import { Col, Row } from 'antd';
+
+// eslint-disable-next-line import/order
```
Comment (enyeInc) [inline src/components/signin/components/FormView.jsx]: do not disable any of the linting rules. Import your code in the correct order

File: src/components/signin/components/PageHelmet.jsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
+// eslint-disable-next-line import/order
```
Comment (enyeInc) [inline src/components/signin/components/PageHelmet.jsx]: same here

File: src/components/signin/components/SignIn.jsx [hunk 8]
```diff
@@ -0,0 +1,13 @@
+/* eslint-disable react/jsx-fragments */
```
Comment (enyeInc) [inline src/components/signin/components/SignIn.jsx]: why is this disabled?

File: src/components/signin/constants.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    charset: 'utf-8',
+}, {
+    metaContent: 'We remain commited to helping you worry less about your laundry. ',
+    metaContentMore: 'Login to continue your stress-free laundry-happy life.',
+    metaName: 'description',
+}, {
+    metaContent: 'width=device-width, initial-scale=1',
+    metaContentMore: '',
+    metaName: 'viewport',
+}];
```
Comment (enyeInc) [inline src/components/signin/constants.js]: the formatting of arrays like this should be as follows
```
const XXXX = [
    {...},
    {...},
    ...
];
```

**Comment by enyeInc** [comment]
@edbella Code looks good, just stop disabling the eslinting rules and a minor array formatting change

File: src/components/signin/components/SignIn.jsx (hunk 8 above)
Comment (edbella) [reply src/components/signin/components/SignIn.jsx]: If I remove it it keeps saying that react fragments aren't supported until react version 16+ And my current version is 16.12. 
I don't know why it keeps saying that.

Comment (edbella) [reply src/components/signin/components/SignIn.jsx]: @enyeInc

Comment (enyeInc) [reply src/components/signin/components/SignIn.jsx]: Not sure what the issue is here, and I do not think its worth the time to dig at the issue. This eslint-disable is fine

File: src/components/signin/components/PageHelmet.jsx [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+    CANON_LINK,
+} = HELMET_STRINGS;
+
+const w = Math.random();
+export default function PageHelmet() {
+    return (
+        <Helmet titleTemplate={TITLE_TEMPLATE} defaultTitle={TITLE_TEMPLATE_DEFAULT_TITLE}>
+            {META_TAGS.map(tag => {
+                const { metaName, metaContent, metaContentMore } = tag;
+                return (<meta key={w} name={metaName} content={metaContent + metaContentMore} />);
```
Comment (enyeInc) [inline src/components/signin/components/PageHelmet.jsx]: why do you have a `metaContent` and `metaContentMore` when you are just going to add it together. you should just have one `content` key associated with each tag.

File: src/components/signin/components/PageHelmet.jsx [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+// De-structuring needed constants from HELMET_STRINGS
+const {
+    TITLE_TEMPLATE,
+    TITLE_TEMPLATE_DEFAULT_TITLE,
+    META_TAGS,
+    PAGE_TITLE,
+    CANON_LINK,
+} = HELMET_STRINGS;
+
+const w = Math.random();
```
Comment (enyeInc) [inline src/components/signin/components/PageHelmet.jsx]: what is this? Do not do this

File: src/components/signin/components/PageHelmet.jsx (hunk 10 above)
Comment (edbella) [reply src/components/signin/components/PageHelmet.jsx]: When I merge it, I get an error saying the line is too long. @enyeInc

Comment (enyeInc) [inline src/components/signin/components/PageHelmet.jsx]: Use the `metaname` as the `key`

Comment (edbella) [reply src/components/signin/components/PageHelmet.jsx]: Beyond the limit of 100 specified in the rules. @enyeInc

Comment (edbella) [reply src/components/signin/components/PageHelmet.jsx]: The same thing here. It tells me the line has passed the 100 length limit @enyeInc

File: src/components/signin/components/PageHelmet.jsx (hunk 11 above)
Comment (edbella) [reply src/components/signin/components/PageHelmet.jsx]: removed

File: src/components/signin/components/PageHelmet.jsx (hunk 10 above)
Comment (enyeInc) [reply src/components/signin/components/PageHelmet.jsx]: that is an easy fix
```
(
   <meta
      key={...}
      ...
   />
)
```

Comment (enyeInc) [reply src/components/signin/components/PageHelmet.jsx]: the 100 line limit is an easy fix

Comment (edbella) [reply src/components/signin/components/PageHelmet.jsx]: I've fixed this. @enyeInc

Comment (edbella) [reply src/components/signin/components/PageHelmet.jsx]: If I use a string literal, another error comes up saying strings must use single quotes. @enyeInc

**Pull request merged**